command/init_test.go

```diff
@@ -410,6 +411,97 @@ func TestInit_backendConfigKV(t *testing.T) {
 	}
 }
 
+func TestInit_backendConfigKVReInit(t *testing.T) {
+	// Create a temporary working directory that is empty
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("init-backend-config-kv"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
+
+	ui := new(cli.MockUi)
+	c := &InitCommand{
+		Meta: Meta{
+			testingOverrides: metaOverridesForProvider(testProvider()),
+			Ui:               ui,
+		},
+	}
+
+	args := []string{"-backend-config", "path=test"}
+	if code := c.Run(args); code != 0 {
+		t.Fatalf("bad: \n%s", ui.ErrorWriter.String())
+	}
+
+	ui = new(cli.MockUi)
+	c = &InitCommand{
+		Meta: Meta{
+			testingOverrides: metaOverridesForProvider(testProvider()),
+			Ui:               ui,
+		},
+	}
+
+	// a second init should require no changes, nor should it change the backend.
+	args = []string{"-input=false"}
```

Comment:
This makes sense, but it makes me wonder then what is the correct way to express "Discard all previously-set `-backend-config` settings"?

Could that be done by setting `-backend-config` to a path to a file containing nothing at all? Or setting `-backend-config` explicitly to the empty string? [Our instructions on the website](https://www.terraform.io/docs/commands/init.html#backend-initialization) don't discuss how you're supposed to express that.

Reply:
That's a good question! We've never ever really had a way to do that.
I think `-reconfigure` should be able to take care of that (though it doesn't currently), meaning that `-reconfigure` with no `-backend-config` will use _only_ the config file.